src/views/preview/preview.jsx

```diff
@@ -73,13 +74,15 @@ class Preview extends React.Component {
             if (this.props.user) {
                 const username = this.props.user.username;
                 const token = this.props.user.token;
+                this.props.getTopLevelComments(this.state.projectId, this.props.comments.length);
                 this.props.getProjectInfo(this.state.projectId, token);
                 this.props.getRemixes(this.state.projectId, token);
                 this.props.getProjectStudios(this.state.projectId, token);
                 this.props.getCuratedStudios(username);
                 this.props.getFavedStatus(this.state.projectId, username, token);
                 this.props.getLovedStatus(this.state.projectId, username, token);
             } else {
+                this.props.getTopLevelComments(this.state.projectId, this.props.comments.length);
```

Comment:
This is probably something to follow up on, but have you looked to see what happens here when you log in or out while on the page? It looks like the transition would cause comments to load? Seems unexpected.

Reply:
Hmm, that's a good point, not sure what would happen (although I agree that's probably a follow up). Might be a reason to switch to another data structure in redux, like a Set (or logic in redux that does the same thing).

Reply:
Oh, I see what you mean (logging in loads the next page of comments, which is... weird). Not sure what would be the right approach for tackling that issue; some sort of pagination state in that component might work?